assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -126,33 +260,93 @@ describe("DiversionPage", () => {
 
     fireEvent.click(originalRouteShape.get(container))
 
-    expect(screen.getByTitle("Detour End")).not.toBeNull()
+    expect(screen.getByTitle("Detour End")).toBeVisible()
   })
 
-  test("when end point has been set, finish detour button is visible", async () => {
+  test("clicking on route shape after the detour is ended doesn't do anything", async () => {
     const { container } = render(<DiversionPage />)
 
     fireEvent.click(originalRouteShape.get(container))
 
-    expect(
-      screen.getByRole("button", { name: "Finish Detour", hidden: true })
-    ).not.toBeVisible()
+    fireEvent.click(container.querySelector(".c-vehicle-map")!)
```

Comment:
issue(non-blocking): I'm working off your branch for the state machine stuff and the tests started failing I _think_ because these aren't wrapped in `act`, I can do so in my PR, but if you're up for it, could you maybe take a pass at the file and wrap fire event's in them?

Reply:
I think that can be arranged. I had been wondering whether `fireEvent` should _always_, _never_, or _sometimes_ be wrapped in `act`.

I'm thinking ... probably _always_?

Reply:
Okay I'm not actually sure that this is the cause, still figuring that out I guess, but I think wrapping fireEvent in act still stands? (unless I'm mis-remembering something

Reply:
I thought always too but I'm looking at the docs and I'm just confused now 🫠

Reply:
Well, I'll do it for now, and see what happens!

Reply:
[Feedback: Wrap all fireEvent calls in act(...)](https://github.com/mbta/skate/pull/2580/commits/7295eebd2d6298a10e9bb0388fa3c40f4a83f471)